tests/exception/t16366.nim

```diff
@@ -0,0 +1,10 @@
+discard """
+  action: run
+  exitcode: 1
+  targets: "c cpp"
```

Comment:
add: `disabled: openbsd`
can be fixed later 
(refs https://builds.sr.ht/~araq/job/369631 which shows
Error: unhandled exception: expected raising 'IOError', instead nothing was raised by: 
)